kubric/core/base.py

```diff
@@ -76,16 +94,24 @@ def active_scene(self):
 
   @tl.default("uid")
   def _uid(self):
-    """ Use the class name and a global count separated by a period as the UID.
+    """Default initializer for uid trait.
+    
+    Use self.name and a global count separated by a period as the UID.
     Counting starts at 1 and always has at least three digits.
     UIDs are thus of the form Scene.001 or Cube.004.
     That way the first 999 instances of each class can be sorted alphabetically.
     """
-    name = self.__class__.__name__
+
+    # TODO(klaus): what is this logic used for?
```

Comment:
"Assets" like `UndefinedMaterial` should be singleton and never have a name (since they are special values for internal use), so I short-circuited the uid creation here to make sure that their uid is always just their class-name.

Reply:
Makes sense, added a comment about it.